demo/TreeViewExamples.jsx

```diff
@@ -51,15 +51,36 @@ export function TreeViewExamples() {
   ];
 
   return (
-    <TreeView
-      nodes={nodes}
-      template={(item, index, parentItem) => (
-        <em>
-          {item.attrA} (index: {index} parent: {parentItem?.attrA || <em>None</em>})
-        </em>
-      )}
-      childrenPath={'children'}
-      draggable={true}
-    />
+    <div>
+      <h1 className="h4">Simple Tree View</h1>
+
+      <TreeView
+        nodes={nodes}
+        template={(item, index, parentItem, path) => (
+          <em>
+            {item.attrA} (index: {index} parent: {parentItem?.attrA || <em>None</em>})
+          </em>
+        )}
+        childrenPath={'children'}
+        draggable={true}
+      />
+
+      <div className="mt-4">
+        <h1 className="h4">Tree View with Form</h1>
+        <Form initialValues={{ nodes }} onSubmit={(newValues) => console.log(newValues)}>
+          <TreeView
+            nodes={nodes}
+            template={(item, index, parentItem, path) => (
+              <em>
+                <p>Path: path</p>
+                <FormInput name={`nodes${path}.attrA`} />
```

Comment:
The point with this change is to allow the usage of a TreeView with Form Inputs. This behavior is not possible using only the 3 args (item, index, parentItem).

![image](https://user-images.githubusercontent.com/29656850/115436403-56265b80-a1e1-11eb-8cad-52036b1dceed.png)